internal/payloads/error_payload.go

```diff
@@ -58,7 +67,7 @@ func NewInvalidRequestError(ctx context.Context, err error) *ResponseError {
 func SourcesClientError(ctx context.Context, message string, err error, status int) *ResponseError {
```

Comment:
couldn't we drop the status and just use NewNotFoundError for the 404?
I think we need to split notFound and badRequest, because those should get reported on differently to the user, 404 is something users can act on and we do not need to log, because support does not need to react, whereas badRequest is the opposite, we need to log it, but to user we just need to send info "Hey our service is broken, try again later"